src/client/app/components/visual-unit/CreateVisualUnitModalMapComponent.tsx

```diff
@@ -0,0 +1,171 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import * as d3 from 'd3';
+import { useEffect } from 'react';
+import { useAppSelector } from '../../redux/reduxHooks';
+import { selectAllUnits } from '../../redux/api/unitsApi';
+import { selectConversionsDetails } from '../../redux/api/conversionsApi';
+
+/**
+ * Visual unit-conversion graph component
+ * @returns D3 force graph visual
+ */
+export default function CreateVisualUnitMapModalComponent() {
+
+	/* Get unit and conversion data from redux */
+	const unitData = useAppSelector(selectAllUnits);
+	const conversionData = useAppSelector(selectConversionsDetails);
+
+	/* Create data container to pass to D3 force graph */
+	const data: { nodes: any[], links: any[] } = {
+		nodes: [],
+		links: []
+	};
+	unitData.map(function (value) {
+		data.nodes.push({'name': value.name,
+			'id': value.id
+		});
+	});
+	conversionData.map(function (value) {
+		data.links.push({
+			'source': value.sourceId,
+			'target': value.destinationId,
+			'bidirectional': value.bidirectional		/* boolean value */
```

Comment:
OED likes comments on the line above and not at the end of the line. Applies to other lines too.